lib/beaker-hiera/helpers.rb

```diff
@@ -9,17 +9,20 @@ module Hiera
         # @param [Host, Array<Host>, String, Symbol] host
         #   One or more hosts to act upon, or a role (String or Symbol) that
         #   identifies one or more hosts.
-        # @param [Array] hierarchy
-        #   One or more hierarchy paths
-        def write_hiera_config_on(host, hierarchy)
+        # @param [Array] paths
+        #   One or more paths within the Hiera data directory
+        def write_hiera_config_on(host, paths)
           block_on host do |hst|
             hiera_config = {
-              backends: 'yaml',
-              yaml: {
-                datadir: hiera_datadir(hst)
+              version: 5,
+              defaults: {
+                datadir: hiera_datadir(hst),
+                data_hash: 'yaml_data',
+              },
+              hierarchy: {
+                name: 'Hiera data',
+                paths: paths,
```

Comment:
`hierarchy` should be an `array` 

Reference: https://www.puppet.com/docs/puppet/7/hiera_config_yaml_5.html#config_syntax

Reply:
But `paths` is already an array.

Reply:
Hmm, I think I understand what you mean:
```
         	Error: Evaluation Error: Error while evaluating a Resource Statement, Lookup of key 'lookup_options' failed: The Lookup Configuration at '/etc/puppetlabs/puppet/hiera.yaml' has wrong type, entry 'hierarchy' expects an Array value, got Struct (file: /tmp/apply_manifest_165059126.pp.wQ3XpE, line: 1, column: 9) on node centos9-64-puppet7.example.com
```